src/ppx/graphics_util.cpp

```diff
@@ -1510,5 +1683,58 @@ Result CreateMeshFromFile(
     return ppx::SUCCESS;
 }
 
+// -------------------------------------------------------------------------------------------------
+
+Result LoadFramesFromRawVideo(
+    const std::filesystem::path&    path,
+    grfx::Format                    format,
+    uint32_t                        width,
+    uint32_t                        height,
+    std::vector<std::vector<char>>* pFrames)
+{
+    PPX_ASSERT_NULL_ARG(pFrames);
+
+    const grfx::FormatDesc* formatDesc = grfx::GetFormatDescription(format);
+    if (formatDesc == nullptr) {
+        PPX_LOG_ERROR("Failed to fetch information for texture format " << format);
+        return ppx::ERROR_FAILED;
+    }
+
+    std::optional<grfx::FormatPlaneDesc> formatPlanes = grfx::GetFormatPlaneDescription(format);
+
+    uint32_t frameSizeInBytes = 0;
+    if (formatPlanes.has_value()) {
+        frameSizeInBytes = GetPlanarImageSizeInBytes(*formatDesc, *formatPlanes, width, height);
+    }
+    else {
+        frameSizeInBytes = formatDesc->bytesPerTexel * width * height;
+    }
+
+    ppx::fs::File file;
+    if (!file.Open(path)) {
+        PPX_ASSERT_MSG(false, "Cannot open the file!");
+        return ppx::ERROR_FAILED;
+    }
+    const size_t size = file.GetLength();
+
+    if (size % frameSizeInBytes != 0) {
+        PPX_LOG_WARN("There may be partial frames in the raw video file at " << path << "; loading as much as possible.");
+    }
+
+    std::vector<char> buffer(frameSizeInBytes);
+    size_t            totalRead = 0;
+    while (totalRead + frameSizeInBytes <= size) {
+        const size_t readSize = file.Read(buffer.data(), size);
+        if (readSize != size) {
+            PPX_ASSERT_MSG(false, "Cannot read the content of the file!");
```

Comment:
The correct message should probably be something like `unable to load video frames: expected X bytes for format F, could only read Y bytes.`

Reply:
Updated